src/components/product/ProductsPage.component.tsx

```diff
@@ -1,49 +1,87 @@
 
+const MOBILE_WIDTH = 900;
+const SCROLL_THRESHOLD = 150;
+const ITEMS_PER_PAGE = 8;
+
 export default function ProductsPage() {
-    const apiUrl = 'https://ma-backend-api.mocintra.com/api/v1/products';
+    const { productsList, isError, isLoading, totalProducts, fetchProducts } = useContext(ProductsDataContext);
     const [products, setProducts] = useState<Product[]>([]);
     const [currentPage, setCurrentPage] = useState(1);
-    const [filters, setFilters] = useState<string[]>([]);
+    const [categoryNumber, setCategoryNumber] = useState<string>('0');
     const [sort, setSort] = useState<string>('newest');
     const [searchQuery, setSearchQuery] = useState<string>('');
 
-    const itemsPerPage = 8;
-
-    const filteredProducts = useFilteredProducts(products, filters);
-    const searchedProducts = useSearchedProducts(filteredProducts, searchQuery);
-    const sortedProducts = useSortedProducts(searchedProducts, sort);
-    const shownProducts = usePaginatedProducts(sortedProducts, currentPage, itemsPerPage);
+    const [isMobile, setIsMobile] = useState(window.innerWidth < MOBILE_WIDTH);
+    const totalPages = Math.ceil(totalProducts / ITEMS_PER_PAGE);
 
     useEffect(() => {
-        getData(apiUrl)
-            .then((data) => {
-                setProducts(data.products);
-            })
-            .catch((error) => console.error(error));
+        const handleResize = () => {
+            setIsMobile(window.innerWidth < MOBILE_WIDTH);
+        };
+
+        window.addEventListener('resize', handleResize);
+        return () => {
+            window.removeEventListener('resize', handleResize);
+        };
     }, []);
 
-    const handleFilterChange = (filter: string) => {
-        setFilters((previousFilters) =>
-            previousFilters.includes(filter) ? previousFilters.filter((f) => f !== filter) : [...previousFilters, filter],
-        );
+    useEffect(() => {
+        fetchProducts({
+            limit: ITEMS_PER_PAGE,
+            offset: (currentPage - 1) * ITEMS_PER_PAGE,
+            categoryId: categoryNumber,
+            title: searchQuery,
+            sortOrder: sort === 'newest' ? 'asc' : 'desc',
+        });
+    }, [currentPage, categoryNumber, sort, searchQuery, fetchProducts]);
+
+    useEffect(() => {
+        if (isMobile) {
+            setProducts((previousProducts) => [...previousProducts, ...productsList]);
+        } else {
+            setProducts(productsList);
+        }
+    }, [productsList, isMobile]);
+
+    const handleFilterChange = (number: string) => {
+        setCategoryNumber(number);
         setCurrentPage(1);
+        setProducts([]);
     };
 
+    const handleScroll = useCallback(() => {
+        if (
+            window.innerHeight + document.documentElement.scrollTop >= document.documentElement.offsetHeight - SCROLL_THRESHOLD &&
+            !isLoading &&
+            currentPage < totalPages
+        ) {
+            setCurrentPage(currentPage + 1);
```

Comment:
Hey, @demasha0908, when this condition is met and a new bunch of products arrives, the scroll jumps to the page top. Please avoid this jumping.

Reply:
Fixed

Reply:
I still see this jump, send you a screen recording in telegram
Resolve this issue, for now as it is not a critical one.